src/components/Login/Login.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import withStyles from 'isomorphic-style-loader/withStyles';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import Cookies from 'universal-cookie';
+import s from './Login.css';
+
+class Login extends React.Component {
+  static propTypes = {
+    oneLine: PropTypes.bool,
+    collapsable: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    oneLine: false,
+    collapsable: false,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      kuski: '',
+      password: '',
+      loggedIn: false,
+      collapsed: true,
+    };
+  }
+
+  componentDidMount() {
+    const cookies = new Cookies();
+    const token = cookies.get('token');
+    const username = cookies.get('username');
+    const userid = cookies.get('userid');
+    if (token && username && userid) {
+      this.setState({
+        kuski: username,
+        loggedIn: true,
+      });
+    }
+  }
+
+  handleChange = name => event => {
+    this.setState({ [name]: event.target.value });
+  };
+
+  login() {
+    const { kuski, password } = this.state;
+    fetch('/token', {
+      method: 'POST',
+      body: JSON.stringify({ kuski, password }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    }).then(response => {
+      response.json().then(body => {
+        const cookies = new Cookies();
+        cookies.set('token', body.Response.token, { path: '/' });
+        cookies.set('username', body.Response.username, { path: '/' });
+        cookies.set('userid', body.Response.userid, { path: '/' });
+        this.setState({ loggedIn: true });
+      });
+    });
+  }
+
+  logout() {
+    const cookies = new Cookies();
+    cookies.remove('token', { path: '/' });
+    cookies.remove('username', { path: '/' });
+    cookies.remove('userid', { path: '/' });
+    this.setState({ loggedIn: false });
+  }
+
+  render() {
+    const { oneLine, collapsable } = this.props;
+    const { kuski, password, loggedIn, collapsed } = this.state;
+    const showForm = !collapsable || !collapsed;
+    return (
+      <>
+        {!showForm && (
+          <Button onClick={() => this.setState({ collapsed: !collapsed })}>
```

Comment:
If you are setting state based on previous state values you should use another variation of setState function, [see the docs](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).

Reply:
wasn't aware of that one actually, fixed in next PR